File: examples/experiments/multilang/nodejs/float_ops.js
function intParam(params, name, defaultValue, min, max) {
    const parsed = Number.parseInt(params?.[name] ?? defaultValue, 10);
    const value = Number.isFinite(parsed) ? parsed : defaultValue;
    return Math.max(min, Math.min(max, value));
}

module.exports = (params, ctx) => {
    params = params || {};

    const iterations = intParam(
        params,
        "iterations",
        6000000,
        100000,
        20000000
    );

    let acc = 0.0;
    let x = 0.123456789;

    for (let i = 1; i <= iterations; i++) {
        x = (
            x * 1.0000001192092896
            + 0.00000035762786865
        ) % 1.0;

        acc +=
            Math.sin(
                x + (i % 31) * 0.001
            )
            * Math.cos(x * 0.5);

        acc +=
            Math.sqrt(x + 1.0)
            / (1.0 + (i % 17));
    }

    return {
        benchmark: "float_ops",
        iterations,
        result: Number(acc.toFixed(9)),
    };
};

File: examples/experiments/multilang/python/float_ops.py
import math


def _int_param(params, name, default, minimum, maximum):
    try:
        value = int(params.get(name, default))
    except (TypeError, ValueError, AttributeError):
        value = default
    return max(minimum, min(maximum, value))


def handler(params, context):
    params = params or {}

    iterations = _int_param(params, "iterations", 6_000_000, 100_000, 20_000_000)

    acc = 0.0
    x = 0.123456789

    for i in range(1, iterations + 1):
        x = (x * 1.0000001192092896 + 0.00000035762786865) % 1.0
        acc += math.sin(x + (i % 31) * 0.001) * math.cos(x * 0.5)
        acc += math.sqrt(x + 1.0) / (1.0 + (i % 17))

    return {
        "benchmark": "float_ops",
        "iterations": iterations,
        "result": round(acc, 9),
    }
